packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -285,6 +285,16 @@ class AssembleCommand extends FlutterCommand {
 
     final Map<String, Object?> defineConfigJsonMap = extractDartDefineConfigJsonMap();
     final List<String> dartDefines = extractDartDefines(defineConfigJsonMap: defineConfigJsonMap);
+    if (results.containsKey(kFlavor) && results[kFlavor]!.isNotEmpty) {
```

Comment:
I think this should be done instead in the ios and macos assemble target steps based on the same logic used to parse configuration and schemes that already exist in xcodeproj, and then from there sent it into the GenSnapshot (if I'm remembering how this works).

Reply:
> should be done instead in the ios and macos assemble target steps

Could you please elaborate on the advantages of this move? I am obviously not an expert here, so I can't see any for now :( But if it is done, I believe there will be code duplication since it will require moving Android's implementation somewhere. Doing the same thing in 2-3 places when it can be done in a single location (as in the current PR state) – I believe that was the main motivation to implement parsing here.

Or maybe I completely misunderstood your comment?